tests/e2e/tests/content-manager/history.spec.ts

```diff
@@ -7,40 +7,71 @@ import { waitForRestart } from '../../utils/restart';
 
 const edition = process.env.STRAPI_DISABLE_EE === 'true' ? 'CE' : 'EE';
 
+const AUTHOR_CREATE_URL =
+  /\/admin\/content-manager\/collection-types\/api::author.author\/create(\?.*)?/;
+const AUTHOR_EDIT_URL =
+  /\/admin\/content-manager\/collection-types\/api::author.author\/(?!create)[^/]/;
+const ARTICLE_CREATE_URL =
+  /\/admin\/content-manager\/collection-types\/api::article.article\/create(\?.*)?/;
+const ARTICLE_LIST_URL = /\/admin\/content-manager\/collection-types\/api::article.article(\?.*)?/;
+const ARTICLE_EDIT_URL =
+  /\/admin\/content-manager\/collection-types\/api::article.article\/[^/]+(\?.*)?/;
+const ARTICLE_HISTORY_URL =
+  /\/admin\/content-manager\/collection-types\/api::article.article\/[^/]+\/history(\?.*)?/;
+const HOMEPAGE_EDIT_URL = /\/admin\/content-manager\/single-types\/api::homepage.homepage(\?.*)?/;
+const HOMEPAGE_HISTORY_URL =
+  /\/admin\/content-manager\/single-types\/api::homepage.homepage\/history(\?.*)?/;
+
 const goToHistoryPage = async (page: Page) => {
-  const moreActionsButton = await page.getByRole('button', { name: /more actions/i });
+  const moreActionsButton = page.getByRole('button', { name: /more actions/i });
   await expect(moreActionsButton).toBeEnabled();
   await moreActionsButton.click();
-  const historyButton = await page.getByRole('menuitem', { name: /content history/i });
-  await historyButton.click();
+  const historyButton = page.getByRole('menuitem', { name: /content history/i });
+
+  // TODO find out why the history button sometimes doesn't appear. Reloading shouldn't be necessary
+  if (await historyButton.isVisible()) {
+    await historyButton.click();
+  } else {
+    await page.reload();
+    await goToHistoryPage(page);
+  }
+};
+
+/**
+ * The success notification lets us make sure an action is completed before moving on.
+ * We also close it to make sure only one appears at a time, otherwise the test fails.
+ */
+const closeSuccessNotification = async (page: Page) => {
+  const successNotification = page.getByRole('alert', { name: /success/i });
+  if (await successNotification.isVisible()) {
+    await successNotification.getByRole('button', { name: /close/i }).click();
+  }
```

Comment:
Not positive but it looks like this util might do the same thing, or is meant to do the same thing...

https://github.com/strapi/strapi/blob/f75e3c6d67cc47c64ab37479efdbb7b43be50b78/tests/e2e/utils/shared.ts#L27-L35